### PR TITLE
feat: popout in header & settings telemetry

### PR DESCRIPTION
**Pull request by dexturr**
## Popout in header

Related to #213 

Allow user to change their telemetry settings in settings

Screenshot 2023-06-26 at 14 53 43

Covering ACs:

- [ ]  There is a way to change whether I want to opt in / out of error reporting later (e.g. in settings)

Move popout button into header:

Screenshot 2023-06-27 at 23 59 24

### CODE REVIEW

File: frontend/components/icons/expand.tsx [hunk 1]
```diff
@@ -0,0 +1,14 @@
+export const ExpandIcon = () => {
+  return (
+    <svg width="16" height="16" viewBox="0 0 16 16" fill="none" xmlns="http://www.w3.org/2000/svg">
```
Comment (mattrussell36) [inline frontend/components/icons/expand.tsx:3]: suggestion (non blocking): I tend to clean out the xmlns attribute. Also, not sure how the icon system is set up here, but I've found removing inline fill/stroke attributes setting all icons to use fill: currentColor a nice way to make the icons more usable

File: frontend/routes/auth/settings/lock-section.tsx [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+  settingsLockButton: 'settings-lock-button',
+  settingsFeedbackDescription: 'settings-feedback-description',
+  settingsFeedbackLink: 'settings-feedback-link'
+}
+
+export const LockSection = () => {
+  const { client } = useJsonRpcClient()
+  const navigate = useNavigate()
+  const { handleSubmit: handleLock } = useForm()
+  const lock = useCallback(async () => {
```
Comment (mattrussell36) [inline frontend/routes/auth/settings/lock-section.tsx:21]: Might be missing full context here, but I feel like some of the useCallback's in this PR might not be needed. https://kentcdodds.com/blog/usememo-and-usecallback#why-is-usecallback-worse

File: frontend/hooks/save-settings/index.tsx [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+import { RpcMethods } from '../../lib/client-rpc-methods'
+import { useGlobalsStore } from '../../stores/globals'
+
+export const useSaveSettings = () => {
+  const [loading, setLoading] = useState(false)
+  const { request } = useJsonRpcClient()
+  const { loadGlobals } = useGlobalsStore((state) => ({
+    loadGlobals: state.loadGlobals
+  }))
+  const save = useCallback(
```
Comment (edd) [inline frontend/hooks/save-settings/index.tsx:12]: **chore (non-blocking)**:  I'd have appreciated some sort of comment on this one, but it's not essential. It probably does provide a hook that persists settings. I don't understand why it needs to `loadGlobals` after saving, but... maybe that's fine?

File: frontend/routes/auth/settings/lock-section.tsx [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+          Lock
+        </Button>
+      </form>
+      <div data-testid={locators.settingsFeedbackDescription}>Spotted any issues or bugs?</div>
+      <ExternalLink
+        data-testid={locators.settingsFeedbackLink}
+        className="text-white mt-1 underline"
+        href={config.feedbackLink}
+      >
+        Provide feedback
```
Comment (edd) [inline frontend/routes/auth/settings/lock-section.tsx:38]: **chore (non-blocking)**: It was only here that I noticed you're not using `t(...)` or similar to make future translation stuff maybe slightly easier maybe. It's consistent so I suppose I'm saying: "enjoy that task when it comes up".

**Review by edd — COMMENTED** [review]
Only comments from me

File: frontend/routes/auth/settings/telemetry-section.spec.tsx [hunk 5]
```diff
@@ -0,0 +1,112 @@
+import { render, screen, fireEvent } from '@testing-library/react'
```
Comment (dalebennett1992) [inline frontend/routes/auth/settings/telemetry-section.spec.tsx:1]: Could we add a test checking that when we toggle the yes/no buttons that they do the following:
1. change to the correct state
2. Only one radio button is ever toggled at one time

Comment (dalebennett1992) [reply frontend/routes/auth/settings/telemetry-section.spec.tsx:1]: I also think the correct hierachy of testing here is this:

unit test:
1. Check we can toggle yes, 
2. Check we can toggle no
3. Check only one radio button selected at either time
4. Check there is always a radio button selected by default(?), as long as the default is consistent

e2e test:

1. Update the configuration in settings
2. refresh
3. check change is retained

Comment (dexturr) [reply frontend/routes/auth/settings/telemetry-section.spec.tsx]: Added unit tests, if you don't mind adding the e2e test 👍

**Review by dalebennett1992 — APPROVED** [review]

**Pull request merged**